R/template.R

```diff
@@ -791,5 +811,78 @@ yardstick_eval_select_dots <- function(...,
 
   out <- names(out)
 
+  set_known_selection(as_label(expr), out)
+
   out
 }
+
+# store known selections (#428) ------------------------------------------------
+is_known_selection <- function(expr_name) {
+  if (!catalog_is_available() || !in_tuning_env()) {
+    return(FALSE)
+  }
+
+  known_selections <- ns_env("tune")$tune_env$known_selections
+
+  expr_name %in% names(known_selections)
+}
+
+get_known_selection <- function(expr_name) {
+  known_selections <- ns_env("tune")$tune_env$known_selections
+
+  known_selections[[expr_name]]
+}
+
+set_known_selection <- function(expr_name, out) {
+  if (!catalog_is_available() || !in_tuning_env()) {
+    return(invisible())
+  }
+
+  tune_env <- ns_env("tune")$tune_env
+
+  if (is.null(tune_env$known_selections)) {
+    init_known_selection(tune_env)
+  }
+
+  tune_env$known_selections[[expr_name]] <- out
```

Comment:
Note:

``` r
boop <- NULL
boop[["a"]] <- 1
boop
#> $a
#> [1] 1
```

<sup>Created on 2023-05-01 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

So no need to initialize `known_selections()`.